CMakeLists.txt

```diff
@@ -312,6 +315,21 @@ if(BUILD_TESTING)
     target_link_libraries(test_repl_str ${PROJECT_NAME})
   endif()
 
+  if(WIN32)
+    set(append_library_dirs "$<TARGET_FILE_DIR:${PROJECT_NAME}>")
+  else()
+    set(append_library_dirs "${CMAKE_CURRENT_BINARY_DIR}")
```

Comment:
Does this need to be a conditional or does `TARGET_FILE_DIR` also work on non-Windows systems?

Reply:
I checked `class_loader` and `pluginlib`. These lines come from there. I tried on my Windows machine and it's the way to make it work.

I will try again to remove this part: `TARGET_FILE_DIR`

Reply:
You will certainly need the generator expression on Windows. I was asking if the same also works on non-Windows so we don't need a conditional choice here.